app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1158,6 +1186,38 @@ private void setupGroupLabel() {
             hasGroupLabel = true;
             FormLayoutHelpers.updateGroupViewVisibility(this, true, shouldHideGroupLabel);
         }
+        updateCompoundIntentButtonVisibility();
+    }
+
+    /**
+     * Identifies whether the questionlist featues an aggregatable intent callout and
+     * displays the appropriate button if so.
+     */
+    private void updateCompoundIntentButtonVisibility() {
+        CompoundIntentList i = this.questionsView.getAggregateIntentCallout();
+        if(i == null) {
+            hideCompoundIntentCalloutButton();
+        } else {
+            Button compoundDispatchButton =
+                    (Button)this.findViewById(R.id.multiple_intent_dispatch_button);
+            compoundDispatchButton.setVisibility(View.VISIBLE);
+            compoundDispatchButton.setText(i.getTitle() + ": " + i.getNumberOfCallouts());
+        }
+    }
+
+    private void hideCompoundIntentCalloutButton() {
+        this.findViewById(R.id.multiple_intent_dispatch_button).setVisibility(View.GONE);
+    }
+
+    private void fireCompoundIntentDispatch() {
+        CompoundIntentList i = this.questionsView.getAggregateIntentCallout();
+        if(i == null) {
+            hideCompoundIntentCalloutButton();
+            Log.e(TAG, "Multiple intent dispatch button shouldn't have been shown");
+            return;
+        }
+
+        this.startActivityForResult(i.getCompoundedIntent(), INTENT_COMPOUND_CALLOUT);
```

Comment:
startActivityForResult has different backstack behavior, mostly. We don't do compound processing yet, but if you fire up with startActivity() on its own the system manages keeping state around in the previous activity differently in a way that can drop more state